control/controlclient/sign_supported.go

```diff
@@ -73,16 +73,27 @@ func findIdentity(subject string, st certstore.Store) (certstore.Identity, []*x5
 			continue
 		}
 
+		if len(chain) == 0 || chain[0] == nil {
+			continue
+		}
+
 		if chain[0].PublicKeyAlgorithm != x509.RSA {
 			continue
 		}
 
 		for _, c := range chain {
+			if c == nil {
+				continue
+			}
 			if c.Subject.String() == subject {
 				selected = id
```

Comment:
how about pulling everything from Line 70 - Line 96 out into a func that can imply "return id" here on match, without so much break/continue flow control? More testable too (should tests get added at some point :)).

Reply:
Fair enough. I was thinking about it but hadn't quite convinced myself it was the best option yet.

Yeah, I was thinking some more about adding tests. I think it'll require forking the certstore package though to add some testing hooks (e.g. an implementation that just loads certs into memory) and to make it build on Linux.

Reply:
I split into a couple functions that made logical sense and that made the flow control less complex. There's probably still some room for improvement in that regard but I think this is well within normal levels of complexity so I'm planning to move on.